make users-guide fails with python >= 3.13

**Describe the bug**
Running `make users-guide` fails with Python versions 3.13 and newer because the `imghdr` module was removed from the standard library after being [deprecated](https://peps.python.org/pep-0594/#deprecated-modules) in Python 3.11.

**To Reproduce**

```
$ make users-guide
...
Extension error:
Could not import extension sphinx.builders.epub3 (exception: No module named 'imghdr')
make[1]: *** [Makefile:40: ../dist-newstyle/doc/users-guide/index.html] Error 2
make[1]: Leaving directory '/home/.../cabal/doc'
make: *** [Makefile:287: users-guide] Error 2
```

**System information**

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 25.04
Release:	25.04
Codename:	plucky
```

I hit this after updating my Ubuntu version. I also hit an earlier problem with `make users-guide-requirements`:

```
$ make users-guide-requirements
make -C doc users-guide-requirements
make[1]: Entering directory '/home/philderbeast/dev/src/cabalism/cabal/doc'
. ../.python-sphinx-virtualenv/bin/activate \
  && pip install --upgrade pip \
  && pip install pip-tools \
  && pip-compile requirements.in
Traceback (most recent call last):
  File "/home/.../cabal/.python-sphinx-virtualenv/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
ModuleNotFoundError: No module named 'pip'
make[1]: *** [Makefile:59: requirements.txt] Error 1
make[1]: Leaving directory '/home/.../cabal/doc'
make: *** [Makefile:291: users-guide-requirements] Error 2
```

I'm not sure, are we meant to handle the venv setup ourselves or is the makefile meant to handle that? I got it to work with:

```
$ python3 -m venv .python-sphinx-virtualenv
$ make users-guide-requirements
...
```

I do have pip:

```
$ pip --version
pip 25.0 from /usr/lib/python3/dist-packages/pip (python 3.13)
```

--- a/.github/workflows/users-guide.yml
+++ b/.github/workflows/users-guide.yml
@@ -44,7 +44,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.10']
+        python-version: ['3.13']
 
     steps:
     - uses: actions/checkout@v6

--- a/.readthedocs.yml
+++ b/.readthedocs.yml
@@ -4,9 +4,9 @@ sphinx:
   configuration: doc/conf.py
 
 build:
-  os: "ubuntu-22.04"
+  os: "ubuntu-24.04"
   tools:
-    python: "3.8"
+    python: "3.13"
   jobs:
     post_create_environment:
       - pip install pip-tools

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -63,9 +63,6 @@ html_theme = 'sphinx_rtd_theme'
 if on_rtd:
     html_style = None
     using_rtd_theme = True
-else:
-    import sphinx_rtd_theme
-    html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
 
 # The name for this set of Sphinx documents.  If None, it defaults to
 # "<project> v<release> documentation".

--- a/doc/requirements.in
+++ b/doc/requirements.in
@@ -1,5 +1,5 @@
-sphinx == 5.3.0
-sphinx_rtd_theme >= 1.2
+sphinx == 8.2.3
+sphinx_rtd_theme >= 3.0.2
 sphinx-jsonschema
 sphinxnotes-strike
 # Pygments>=2.7.4 suggested by CVE-2021-20270 CVE-2021-27291
@@ -14,5 +14,3 @@ jinja2 >= 3.1.4
 idna >= 3.7
 # CVE-2024-35195
 requests >= 2.32.0
-# https://github.com/haskell/cabal/issues/11246
-docutils <0.18

